pkg/gateway/ingress.go

```diff
@@ -76,4 +82,22 @@ func getIngressSpec(manager *v1alpha1.CheManager) *v1beta1.Ingress {
 			},
 		},
 	}
+
+	if manager.Spec.TlsSecretName != "" {
+		ingress.Spec.TLS = []v1beta1.IngressTLS{
+			{
+				Hosts:      []string{manager.Spec.GatewayHost},
+				SecretName: manager.Spec.TlsSecretName,
```

Comment:
so how the secret is copied to dw namespace?

Reply:
This concrete piece of code is responsible for the ingress for the gateway. The gateway lives in the namespace of the CheManager instance, where the secret also resides by requirement.

Reply:
so dw endpoints are created directly by DWO? And DWO is responsible to copy the secret as well?

Reply:
dw endpoints are handled in a different part of code: dw endpoints are handled in https://github.com/che-incubator/devworkspace-che-operator/pull/38/files#diff-8849eb3a15e24de014b7c47930e742a0212f2417dc13a784dfdb7281e16e0483R32 and below.